Fix part of #1824 : Full UI: Tablet: StoryActivity Landscape + Portrait [WIP] [DO NOT MERGE]

prayutsu opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fix part of #1824 
Added layouts for landscape version of tablet layouts and matched UI with the mock

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

prayutsu committed:
made layout changes

prayutsu committed:
added layout files for landscape and matched ui with the mock

prayutsu commented:
@rt4914 PTAL

rt4914 commented:
@prayutsu You have mentioned that you have "matched UI with the mock" , but your screenshot do not match with the mock. So what exactly you are trying to mention here. Thanks.

prayutsu commented:
> @prayutsu You have mentioned that you have "matched UI with the mock" , but your screenshot do not match with the mock. So what exactly you are trying to mention here. Thanks.

I am trying to mention that by changing padding, the size of the recyclerview item is matching with that in the mock.

rt4914 commented:
> > @prayutsu You have mentioned that you have "matched UI with the mock" , but your screenshot do not match with the mock. So what exactly you are trying to mention here. Thanks.
> 
> I am trying to mention that by changing padding, the size of the recyclerview item is matching with that in the mock.

Okay, that will actually be verified in the final implementation of highfi work. Highfi work should not be done in parts like this, instead everything related to highfi should be in a single PR for a particular screen/component. 

This can be merged now, right?

prayutsu commented:
> Okay, that will actually be verified in the final implementation of highfi work. Highfi work should not be done in parts like this, instead everything related to highfi should be in a single PR for a particular screen/component.
> 
> This can be merged now, right?

Can you please summarize once, what else needs to be done to close #1824?

anandwana001 committed:
Fix part of #973 : Fix app module tests for Roboelectric in AddProfileActivityTest (#1842)

* fix AddProfileActivityTest for robolectric

* added todo

anandwana001 committed:
fix AdminAuthActivityTest for robolectric (#1852)

anandwana001 committed:
Fix part of #973 : Fix app module tests for Roboelectric in PinPasswordActivityTest (#1854)

* fix PinPasswordActivityTest for robolectric

* added todo

prayutsu committed:
added custom vertical line

anandwana001 committed:
fix ProfileResetPinActivityTest for robolectric (#1865)

anandwana001 committed:
Fix part of #973 : Fix app module tests for Roboelectric in ExplorationActivityTest  (#1850)

* fix ExplorationActivityTest for robolectric

* max line fix

* nit changes

* aaded todo 1855

anandwana001 committed:
Fix part of #973 : Fix app module tests for Roboelectric in OngoingTopicListActivityTest (#1863)

* fix OngoingTopicListActivityTest test for robolectric

* removed unnecessary functions

anandwana001 committed:
fix ProfileListFragmentTest for robolectric (#1864)

prayutsu committed:
save changes

prayutsu committed:
Fix #1840 : Introduce central utility file for append text in testing (#1853)

* created testUtilFile

* testCoroutineDispatchers causing problem

* improved testUtils

* Imported appendText to all the files

* Made reviewed changes

* made reviewed channges

* updated PinPasswordActivityTest.kt

* update latest files

rt4914 commented:
> > Okay, that will actually be verified in the final implementation of highfi work. Highfi work should not be done in parts like this, instead everything related to highfi should be in a single PR for a particular screen/component.
> > This can be merged now, right?
> 
> Can you please summarize once, what else needs to be done to close #1824?

@prayutsu Any updates on this?

BenHenning committed:
Introduce data provider extension functions (#1836)

* Move DataProvider helper functions to be extension functions, instead,
for improved readability.

Migrates most LiveData returning methods in the domain layer to instead
return DataProviders since toLiveData() is conveniently an extension
function now.

Fixes a bunch of classes whose comment & docs extend past the column
limit.

* Fix (most) app module tests.

* Fix data module tests.

* Fix testing tests.

* Fix utility tests.

* Fix most domain tests.

* Lint fixes.

* Fix broken tests.

This included adding a few new synchronization points since certain
profile behavior was added to a DataProvider, and reimplementing test
application arrangement in some tests since the application is managed
differently now that most tests are using the extend-application
arrangement setup.

BenHenning committed:
Fix #560: Create Oppia Android README (#1722)

* Create Oppia Android README

* s/Oppia/Oppia Android/ in the title

* Update README.md

Address review comments.

* Update README.md

Address reviewer comments: add link for coding contribution instructions, and add badge for newcomer issues.

BenHenning committed:
Move all files to their correct new location. (#1879)

This moves all Kotlin source files to be under an org.oppia.android
structure, but does not actually change references. This is done in two
steps to try and preserve history, and to simplify the code review.

THIS WILL BREAK THE CODEBASE WHEN CHECKED IN.

BenHenning committed:
Fix #1833: Rename package to org.oppia.android (part 2/2) (#1876)

* Move all source files to new org/oppia/android directory structure. No
actual package or code references are updated yet.

* Update Kotlin & XML references to match new package structure.

* Migrate codepaths for Bazel builds to new package structure.

* Fix package references in Gradle files for new package structure.

* Update package & pathing references to new structure for proto files.

* Rename Android package to org.oppia.android.

This includes introducing a new developer Firebase project. It also
fixes references to Android resource R file imports & databinding since
they both exist at the package level rather than the app module level.

* Fix Bazel builds for new package structure.

This required some changes to test setup/arrangement in Bazel, and some
existing issues needed to be fixed (missing dependencies for work
manager, switching multidex, one renamed test, incorrect module
references in tests, and files now referencing resources needing to be
moved).

This includes some Bazel file reformatting that happened automatically
by my Bazel file plugin.

* Resolve merge conflicts.

This won't compile due to duplicate imports which requires a ktlint
clean-up. That will happen in a subsequent commit since it will likely
pick up unrelated changes.

* Remove some merge conflict markers that were missed.

* Remove redundant commits & other ktlint updates.

* Fix incorrect references to new testing package location.

* Post-testing module cleanups via ktlint.

* More post-merge fixes.

Fix new references to R and databinding. Move EditTextInputAction to
correct package.

* Move some missed files that only caused breakages in Bazel.

* Move all files to their correct new location.

This moves all Kotlin source files to be under an org.oppia.android
structure, but does not actually change references. This is done in two
steps to try and preserve history, and to simplify the code review.

THIS WILL BREAK THE CODEBASE WHEN CHECKED IN.

* Fix missed files from the migration after merging.

* Disable Firebase data collection.

* Update BUILD.bazel

Capitalize TODO comment for consistency with other TODO comments.

anandwana001 commented:
@prayutsu You need to update the branch with `develop` before working more on this PR as we had updated the package names.

BenHenning committed:
Improve badges in README (#1905)

prayutsu commented:
> @prayutsu You need to update the branch with `develop` before working more on this PR as we had updated the package names.

Okay, I'll update it

prayutsu committed:
save changes

prayutsu committed:
rebased branch

prayutsu committed:
complete rebase

prayutsu committed:
Merge branch 'develop' of https://github.com/oppia/oppia-android into make-ui-changes-in-tablet-layouts

# Conflicts:
#	app/src/main/res/layout-sw600dp-land/story_chapter_view.xml

prayutsu committed:
complete rebase

prayutsu committed:
update fork

prayutsu commented:
@rt4914 I tried making a line (currently roughly similar to the one in mock). I am stuck on making it of the desired height. 

(Please refer to the self-reviewed comment)

If I set the ```layout_height``` to ```match_parent``` then the following result is obtained - 
![Screenshot_1601142183](https://user-images.githubusercontent.com/54636525/94347062-d2433100-004e-11eb-8c14-6ffcb2b2d7f1.png)

prayutsu commented:
If I set the ```layout_height``` to ```200dp``` then the following result is obtained - 

![Screenshot_1601142271](https://user-images.githubusercontent.com/54636525/94347063-d4a58b00-004e-11eb-9264-95e7ec96590a.png)

prayutsu commented on app/src/main/res/layout-sw600dp-port/story_chapter_view.xml:
```diff
@@ -0,0 +1,142 @@
+        <org.oppia.android.app.customview.VerticalDashedLine
+          android:layout_width="5dp"
+          android:id="@+id/line"
+          android:layout_height="200dp"
```
If I am setting the layout_height here to match_parent then it's not working as expected.

prayutsu commented:
Why ```onMeasure``` is not working properly?

prayutsu commented:
Also, it does not show any views in the design mode fir ```story_chapter_view``` which makes it really difficult to see the results because every time I make some changes, I have to fire up the emulator again. Can this be fixed?
![Screenshot from 2020-09-26 23-11-09](https://user-images.githubusercontent.com/54636525/94347391-58f90d80-0051-11eb-8813-e0a1c229a88c.png)

anandwana001 commented:
Try this [reference](https://stackoverflow.com/questions/55135893/how-to-implement-vertical-dotted-line-below-image-in-android)

prayutsu commented:
@anandwana001 The reference contains really a beautiful example but we want "rounded corners" of the dashes of the line and this reference contains simple rectangular dashes.
https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/d70a3ade-826a-4ac6-95e6-907a24d7b896/

anandwana001 commented:
> @anandwana001 The reference contains really a beautiful example but we want "rounded corners" of the dashes of the line and this reference contains simple rectangular dashes.
> https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/screen/d70a3ade-826a-4ac6-95e6-907a24d7b896/

That something we can modify and as I mentioned, you can check this for just a reference.

rt4914 commented:
@anandwana001 If this PR is ready from your side, should I review it?

anandwana001 commented:
> @anandwana001 If this PR is ready from your side, should I review it?
> 

I had a discussion with the @prayutsu regarding this. It will be good if you give a look, although this PR is WIP.

rt4914 commented:
@prayutsu Any updates after our discussion?
